testing/regress/ecl/parquet_types.ecl

```diff
@@ -0,0 +1,397 @@
+/*##############################################################################
+    HPCC SYSTEMS software Copyright (C) 2017 HPCC Systems®.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+        http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+##############################################################################*/
+
+//class=parquet
+//nothor,noroxie
+//Cover's data type's supported by ECL and arrow
+
+IMPORT Parquet;
+
+//All covered ECL data type's tested below
+
+// BOOLEAN type
+ParquetIO.write(DATASET([
+    {000, 'aaa', TRUE},
+    {001, 'aab', FALSE},
+    {002, 'aac', TRUE},
+    {003, 'aad', FALSE},
+    {004, 'aae', TRUE}
+], {UNSIGNED testid, STRING3 testname, BOOLEAN value}), '/var/lib/HPCCSystems/mydropzone/BooleanTest.parquet', TRUE);
+
+booleanDataset := ParquetIO.Read({UNSIGNED testid; STRING3 testname; BOOLEAN value}, '/var/lib/HPCCSystems/mydropzone/BooleanTest.parquet');
+booleanResult := IF(COUNT(booleanDataset) = 5, 'Pass', 'Fail: Boolean data count mismatch');
```

Comment:
Comparing Row counts won't effectively test for changes in the output. Consider a change in how a string is converted to a boolean when that field is defined in ECL. If that change introduced an error in the translation it wouldn't affect the row count, but it would change the output from reading the file.

Instead of comparing the count of rows to a fixed number, compare the data that is read in to the temporary dataset you defined when writing the data to the parquet file i.e. instead of 
```
ParquetIO.write(DATASET([
    {000, 'aaa', TRUE},
    {001, 'aab', FALSE},
    {002, 'aac', TRUE},
    {003, 'aad', FALSE},
    {004, 'aae', TRUE}
], {UNSIGNED testid, STRING3 testname, BOOLEAN value}), '/var/lib/HPCCSystems/mydropzone/BooleanTest.parquet', TRUE);

booleanDataset := ParquetIO.Read({UNSIGNED testid; STRING3 testname; BOOLEAN value}, '/var/lib/HPCCSystems/mydropzone/BooleanTest.parquet');
booleanResult := IF(COUNT(booleanDataset) = 5, 'Pass', 'Fail: Boolean data count mismatch');
```

Compare the data in the rows:

```
booleanDataIn := DATASET([
    {000, 'aaa', TRUE},
    {001, 'aab', FALSE},
    {002, 'aac', TRUE},
    {003, 'aad', FALSE},
    {004, 'aae', TRUE}
], {UNSIGNED testid, STRING3 testname, BOOLEAN value});
ParquetIO.write(booleanDatasetOut, '/var/lib/HPCCSystems/mydropzone/BooleanTest.parquet', TRUE);

booleanDatasetIn := ParquetIO.Read({UNSIGNED testid; STRING3 testname; BOOLEAN value}, '/var/lib/HPCCSystems/mydropzone/BooleanTest.parquet');
booleanResult := IF(booleanDatasetIn == booleanDatasetOut, 'Pass', 'Fail: Boolean data mismatch');
```

I'm not sure of the best way to compare the data in the datasets and that equality may not work, but you need to make sure what is being read hasn't changed rather than the count of the rows changing.